dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityModeHandler.java

```diff
@@ -0,0 +1,70 @@
+package com.dotcms.rendering.velocity.servlet;
+
+import com.dotcms.rendering.velocity.services.VelocityType;
+import com.dotcms.rendering.velocity.util.VelocityUtil;
+import com.dotcms.visitor.business.VisitorAPI;
+
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.web.HostWebAPI;
+import com.dotmarketing.business.web.WebAPILocator;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotRuntimeException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.portlets.htmlpageasset.model.IHTMLPage;
+import com.dotmarketing.util.Config;
+import com.dotmarketing.util.PageMode;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.StringWriter;
+import java.io.Writer;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.apache.velocity.Template;
+
+public abstract class VelocityModeHandler {
+    
+    protected String CHARSET = Config.getStringProperty("CHARSET", "UTF-8");
+    protected static HostWebAPI hostWebAPI = WebAPILocator.getHostWebAPI();
+    protected static VisitorAPI visitorAPI = APILocator.getVisitorAPI();
+    
+    
+    
+    abstract void serve() throws Exception;
+
+    abstract void serve(Writer out) throws DotDataException, IOException, DotSecurityException;
+
+
+    public final String eval() {
+        StringWriter out = new StringWriter(4096);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 4096 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS109)